packages/components/src/internal/components/domainproperties/DomainForm.tsx

```diff
@@ -334,40 +354,47 @@ export class DomainFormImpl extends React.PureComponent<IDomainFormInput, IDomai
         // if this domain has any Ontology Lookup field(s), check if we need to update the related field properties
         // based on the updated domain (i.e. check for any name changes to selected fields)
         // note: we skip any rowIndexChange which has a newIndex as those are just reorder changes
-        if (rowIndexChange?.newIndex === undefined && ontologyLookupIndices.length > 0) {
-            ontologyLookupIndices.forEach(index => {
-                // skip any ontology lookup fields if they were removed
-                const ontFieldRemoved = rowIndexChange?.originalIndex === index;
-
-                if (!ontFieldRemoved) {
-                    updatedDomain = updateOntologyFieldProperties(
-                        // check for a field removal prior to the ontology lookup field
-                        rowIndexChange?.originalIndex < index ? index - 1 : index,
-                        domainIndex,
-                        updatedDomain,
-                        domain,
-                        rowIndexChange?.originalIndex
-                    );
+        if (rowIndexChanges) {
+            rowIndexChanges.forEach((rowIndexChange) => {
+                if (rowIndexChange?.newIndex === undefined && ontologyLookupIndices.length > 0) {
+                    ontologyLookupIndices.forEach(index => {
+                        // skip any ontology lookup fields if they were removed
+                        const ontFieldRemoved = rowIndexChange?.originalIndex === index;
+
+                        if (!ontFieldRemoved) {
+                            updatedDomain = updateOntologyFieldProperties(
+                                // check for a field removal prior to the ontology lookup field
+                                rowIndexChange?.originalIndex < index ? index - 1 : index,
+                                domainIndex,
+                                updatedDomain,
+                                domain,
+                                rowIndexChange?.originalIndex
+                            );
+                        }
+                    });
                 }
             });
         }
 
         this.setState(() => ({ fieldDetails: updatedDomain.getFieldDetails() }));
 
-        this.props.onChange?.(updatedDomain, dirty !== undefined ? dirty : true, rowIndexChange);
+        this.props.onChange?.(updatedDomain, dirty !== undefined ? dirty : true, rowIndexChanges);
     }
 
     onDeleteConfirm(index: number) {
```

Comment:
I considered consolidating onDeleteConfirm and onBulkDeleteConfirm since they're similar, but looking at the structure decided that might be confusing enough to not merit the change

Reply:
Done!